src/zip_node.js

```diff
@@ -1,16 +1,73 @@
 const { Buffer } = require('buffer')
 const fs = require('fs')
-const { basename, extname, normalize, sep } = require('path')
+const os = require('os')
+const { basename, extname, join, normalize, sep } = require('path')
 const { promisify } = require('util')
 
+const cpFile = require('cp-file')
+const makeDir = require('make-dir')
+const pMap = require('p-map')
 const unixify = require('unixify')
 
 const { startZip, addZipFile, addZipContent, endZip } = require('./archive')
+const { ARCHIVE_FORMAT_ZIP } = require('./utils/consts')
 
 const pStat = promisify(fs.stat)
+const pWriteFile = promisify(fs.writeFile)
 
-// Zip a Node.js function file
-const zipNodeJs = async function ({ basePath, destPath, filename, mainFile, pluginsModulesPath, aliases, srcFiles }) {
+// Taken from https://www.npmjs.com/package/cpy.
+const COPY_FILE_CONCURRENCY = os.cpus().length === 0 ? 2 : os.cpus().length * 2
```

Comment:
Could `cpy` be used directly?
One potential issue is the globbing, maybe it can be disabled?

Reply:
I tried using `cpy` initially, but I wasn't able to correctly preserve the hierarchy of files in the target directory.
The library offers the [`parents` option](https://github.com/sindresorhus/cpy#parents), but it doesn't seem to do exactly what we need, i.e. it doesn't seem to compute the relative path like we're doing.

Reply:
I've had the same problem pretty often when using that library. The `parents` option is not intuitive.
Usually, I manage to copy recursively using the following (odd) syntax:

```js
await cpy('**', destDir, { cwd: srcDir, parents: true })
```

Reply:
Yes, I've considered that. We're using that approach somewhere in the ZISI tests.

However, I don't think it'll work here because we don't want to copy all files under a certain directory. We only want to copy the files contained in the array.

Also, because of the `aliases` object, we need the ability to modify individual paths before copying them, which we wouldn't be able to do using `cpy`.

Finally, `cpy` is currently under `devDependencies`, and by moving it to `dependencies` we'd be adding to the package size. I would rather avoid that, if possible.

Reply:
This makes sense, thanks for explaining :+1: